appcues/src/main/java/com/appcues/data/model/Step.kt

```diff
@@ -3,12 +3,14 @@ package com.appcues.data.model
 import com.appcues.trait.BackdropDecoratingTrait
 import com.appcues.trait.ContainerDecoratingTrait
 import com.appcues.trait.MetadataSettingTrait
+import com.appcues.trait.PresentingTrait
 import com.appcues.trait.StepDecoratingTrait
 import java.util.UUID
 
 internal data class Step(
     val id: UUID,
     val content: ExperiencePrimitive,
+    val presentingTrait: PresentingTrait,
```

Comment:
every step contains presentingTrait to make it easier to reference it later (same way we have contaienrDecoratingTrait here)

Reply:
gotta think of a better way of referencing the presenting trait that is easier to understand and testable then. because the way it was before we had to consult a list to find the containerId, and then use the containerId to get the presentingTrait from ANOTHER list, there was always a missing branch that was impossible to test because it was impossible to reach, it usually means that we can simplify code.

Reply:
I think thats the argument that the state machine doesnt know about "container", it only knows about steps, I guess this is the reason why it feels easier to put it inside the step....